examples/platform/esp32/shell_extension/openthread_cli_register.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "sdkconfig.h"
+#if CONFIG_OPENTHREAD_CLI
+#include "esp_check.h"
+#include "esp_err.h"
+#include "esp_log.h"
+#include "freertos/FreeRTOS.h"
+#include "openthread_cli_register.h"
+#include "platform/ESP32/OpenthreadLauncher.h"
+#include <lib/shell/Engine.h>
+
+#define CLI_INPUT_BUFF_LENGTH 256u
+namespace chip {
+
+static CHIP_ERROR OtcliHandler(int argc, char * argv[])
+{
+    /* the beginning of command "matter esp ot cli" has already been removed */
+    char * cli_str = (char *) malloc(CLI_INPUT_BUFF_LENGTH);
```

Comment:
could we use some RAII allocation (like unique_ptr) to handle frees and have explicit calls on when we need to not free anymore?

Reply:
Ok, unique_ptr is adopted to store the string and trans it to cli_transmit_task.